backend/src/utils/rag.py

```
@@ -0,0 +1,247 @@
+from langchain_openai import ChatOpenAI
+from pydantic import BaseModel, Field
+from typing import Optional, List
+
+from pydantic import BaseModel, Field
```

Ruff (0.14.14)

[error] Redefinition of unused `BaseModel` from line 2: `BaseModel` redefined here

Remove definition: `BaseModel`

(F811)